app/fruits/page.tsx

```diff
@@ -1,43 +1,40 @@
-import { cookies } from 'next/headers';
 import Link from 'next/link';
 import { fruits } from '../../database/fruits';
-import { CookieValue } from '../../util/cookies';
+import { getCookieByName } from '../../util/cookies';
 
 export const metadata = {
   title: 'Fruits',
   description: 'This is my Fruits Page',
 };
 
+// we add this only if we have no dynamic function as cookies or headers
+export const dynamic = 'force-dynamic';
+export const revalidate = false;
+
 export default function FruitsPage() {
   // get the cookie from the server
-  const fruitsCookie = cookies().get('fruitsCookie');
-
-  // create a default value if cooke doesn't exist
-  let fruitsCookieParsed: CookieValue = [];
-
-  if (fruitsCookie) {
-    fruitsCookieParsed = JSON.parse(fruitsCookie.value);
-  }
+  const userFruitCookie = getCookieByName('fruitNotes');
+  const fruitNotes = Array.isArray(userFruitCookie) ? userFruitCookie : [];
 
-  const fruitsWithStars = fruits.map((fruit) => {
-    const fruitWithStars = { ...fruit, stars: 0 };
+  const fruitsWithNotes = fruits.map((fruit) => {
+    const fruitWithNote = { ...fruit, note: '' };
```

Comment:
if you avoid this variable and move the object literal down to the `return`, that probably would be simpler code

(I'm trying to make it easier for students to understand the code)

if you think that the simpler code is harder for them to understand, then leave it as is